File: src/clipboard.ts
import clipboard from 'clipboardy'

export async function copyToClipboard(text: string): Promise<void> {
  try {
    await clipboard.write(text)
  } catch (error: unknown) {
    const message =
      error instanceof Error ? error.message : 'Unknown clipboard error'
    throw new Error(`Failed to copy prompt to clipboard: ${message}`)
  }
}

File: src/clipboard.rs
use arboard::Clipboard;

pub fn copy_to_clipboard(text: &str) -> anyhow::Result<()> {
    let mut clipboard =
        Clipboard::new().map_err(|e| anyhow::anyhow!("Failed to copy prompt to clipboard: {e}"))?;
    clipboard
        .set_text(text)
        .map_err(|e| anyhow::anyhow!("Failed to copy prompt to clipboard: {e}"))?;
    Ok(())
}
